compiler/noirc_driver/src/lib.rs

```diff
@@ -367,6 +367,48 @@ pub fn compile_main(
     Ok((compiled_program, warnings))
 }
 
+pub fn formal_verify_main(
```

Comment:
This function is identical to compile_main.

Currently, it is not clear why you copy it.

The follow-up PR #90 sheds no light either.

Explain, please.

Reply:
It's not identical to `compile_main` because the error message at line 382 differs from the error message at line 340. The general idea of defining a separate function is because in the near future we will be defining a function which will be similar to `compile_no_check`. This function will be called in the body of `formal_verify_main`. We want the logic of compilation and formal verification to be separate to some degree.

Another reason for copying `compile_main` is the ability to have different error reporting for formal verification.

Perhaps it is too soon to add `formal_verify_main` at the current stage of our prototype.

Reply:
> It's not identical to `compile_main` because the error message at line 382 differs from the error message at line 340. 

I noticed this, but I wouldn't say this is a significant difference :-)

> Perhaps it is too soon to add `formal_verify_main` at the current stage of our prototype.

That's what I was thinking too. Since it's just copied code (with a small change in the message), let's introduce it at a later stage. It will be clearer then what's the purpose of the function.

Please, remove and use `compile_main` for now.

Reply:
Done.